protocols/v2/codec-sv2/src/lib.rs

```diff
@@ -1,3 +1,55 @@
+//! # Stratum V2 Codec Library
+//!
+//! This crate provides the encoding and decoding functionality for the Stratum V2 protocol,
+//! handling secure communication between clients and servers.
+//!
+//! `codec-sv2` is essential for implementing the Sv2 protocol correctly and securely. It abstracts
+//! the complexity of message encoding, decoding, and encryption, providing a reliable and
+//! consistent foundation for mining software imperative for interoperability.
+//!
+//!
+//! ## Modules
+//!
+//! * `decoder`: Provides functionality to decode Stratum V2 messages.
+//! * `encoder`: Provides functionality to encode Stratum V2 messages.
+//! * `error`: Defines error types and result aliases used throughout the crate.
+//!
+//! ## Usage
+//!
+//! This crate is designed to be used in mining software that needs to communicate securely
+//! using the Stratum V2 protocol. It supports optional Noise protocol features for encryption
+//! and ensures data integrity and confidentiality.
+//!
+//! ## Example
+//!
+//! ```ignore
+//! use codec_sv2::{State, HandshakeRole, NoiseCodec};
+//! use noise_sv2::Initiator;
+//!
+//! // Initialize the codec state
+//! let role = HandshakeRole::Initiator(Box::new(Initiator::new()));
+//! let mut state = State::not_initialized(&role);
+//! state = State::initialized(role);
+//!
+//! // Proceed with the handshake steps
+//! match state.step_0() {
+//!     Ok(frame) => {
+//!         // Send the initial handshake frame
+//!     }
+//!     Err(e) => {
+//!         // Handle error
+//!     }
+//! }
+//! ```
```

Comment:
Here, we can have a doc test.

Reply:
this could be good for CI, but will require more research to figure out the best way to achieve this

Reply:
When we run `cargo test`, all the unit tests and doc tests are executed, and our CI already includes them.